### PR TITLE
feat: [IA-729] Migration UI, loader, messages

### PR DESCRIPTION
**Pull request by pietro909**
## Short description
This PR adds a trigger for the migration in `MessagesHomeScreen` and tracks the migration status via `allPaginated.migration` property.
Upon completion, the `messagesStatus` store is cleaned up an left unused.

## List of changes proposed in this pull request
- Kick off the migration [right from the view](https://github.com/pagopa/io-app/pull/3853/files#diff-c33626b053f37eaec7a3d3759360ed768a2abfebc99664c890d346f8c779c602R142)
- Add an _ad-hoc_ component [MigratingMessage](ts/screens/messages/paginated/MigratingMessage.tsx) to handle loader and errors
- Add [migrateToPaginatedMessages](https://github.com/pagopa/io-app/pull/3853/files#diff-5b727d84caebb631cf6843fbd8f3595a97801341cee749b8d9535aabdbc12460R180) async action and [resetMigationStatus](https://github.com/pagopa/io-app/pull/3853/files#diff-5b727d84caebb631cf6843fbd8f3595a97801341cee749b8d9535aabdbc12460R189) to be used only in the context of the migration, and [handle them](https://github.com/pagopa/io-app/pull/3853/files#diff-482523dcf54c84f4e40e09c47ee2eeca62c92423ef82ff98aae91839199d8c16R85)
- Add the [MigrationStatus](https://github.com/pagopa/io-app/pull/3853/files#diff-482523dcf54c84f4e40e09c47ee2eeca62c92423ef82ff98aae91839199d8c16R35) to the `allPaginated` reducer to track the migration
- the `migrateToPagination` helper required [extensive rework](https://github.com/pagopa/io-app/pull/3853/files#diff-bcbf6afc3d95f5b125321b43a35962fc8b5fa5186b29df65796d739bd9113817R36) to be used correctly in the saga

### FAQs
1. **I really don't like the message/icon/font you picked for the migration, can you change it?** I tried my best without input. Please suggest anything better.
2. **Where are the tests for the new stuff?** Kindly check [IA-753](https://pagopa.atlassian.net/browse/IA-753).
3. **Where is the analytics?** Please see [IA-754](https://pagopa.atlassian.net/browse/IA-754)
4. **Why aren't you using workers for so many requests?** Have a look at our [Release Plan](https://pagopa.atlassian.net/wiki/spaces/IOAPP/pages/474349926/Rilascio+in+produzione)
5. **Where is the toilet?** I'm afraid I cannot help you with that unless of course you come over here and use mine.

### In production

Note: please check with the BE team whether the API is actually available in production.

1. install the current release of the app
4. mark a few messages as _archived_ and open some of them to mark them _read_
6. note down the messages to compare them with the final result
7. install this branch on the device
8. on the first run, the migration starts
9. after a successful migration you shall see the new UI
10. check against the notes taken in (3) to verify that the process is working
11. no more migration must be done upon further reboots

### In dev

I suggest you update the dev-server to add some delay in the response and/or random errors to test for edge cases.

Add the snippet below in `ts/App.tsx` right before the `App` declaration:

```typescript
setTimeout(() => {
  const messages =
    store.getState().entities.messages.allPaginated.inbox.data.value?.page ||
    [];
  messages.forEach(message => {
    console.log(message.id);
    store.dispatch(DEPRECATED_setMessageReadState(message.id, true, "unknown"));
    store.dispatch(DEPRECATED_setMessagesArchivedState([message.id], true));
  });
}, 10000);
```

In `ts/screens/messages/paginated/MessagesHomeScreen.tsx` modify the condition on line 139 to always return `false`:

```typescript
const needsMigration = Object.keys(messagesStatus).length > 1000;
```

Open the app, log in, scroll down a few pages in the Inbox. After 10s the snippet above will mark all the messages it finds in Inbox as _read_ and _archived_. Take note of the IDs. 
Close the app, comment out the snippet, remove the patch of `needsMigration`, and start again the app.
Upon login, you should see the migration start and then finish.

The test in the video below was conducted locally with 68 messages and a 600ms delay per request. The Archived tab has much more pages than Inbox and many messages are marked as _read_ as a result of the migration described above.

https://user-images.githubusercontent.com/2094604/159877568-503af4b5-4a93-4d84-8e32-1a311e786399.mov

In the following recording, a failing message:

https://user-images.githubusercontent.com/2094604/159877780-12890f3b-9a84-4d65-b814-12c09cff5820.mov

### CODE REVIEW

**Comment by thisisjp** [comment]
@pietro909 some minor fixes if possible:

1. The font used in the retry button is wrong
2. Can we use `pictograms/payment-completed.svg` for the happy case and `pictograms/doubt.svg` for the failure case?

File: ts/screens/messages/paginated/MigratingMessage.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  status: MigrationStatus;
+  onRetry: () => void;
+  onEnd: () => void;
+};
+
+const MigratingMessage = ({ status, onRetry }: Props) => {
+  useEffect(() => {
+    // eslint-disable-next-line no-underscore-dangle
+    if (status.toUndefined()?._tag === "succeeded") {
+      setTimeout(onRetry, 2000);
```
Comment (emiliopavia) [inline ts/screens/messages/paginated/MigratingMessage.tsx:44]: Please can you explain this timeout? Thanks!

**Review by emiliopavia — CHANGES_REQUESTED** [review]

File: ts/screens/messages/paginated/MigratingMessage.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  },
+  activityIndicator: {
+    padding: 12
+  }
+});
+
+type Props = {
+  status: MigrationStatus;
+  onRetry: () => void;
+  onEnd: () => void;
```
[the remark this answers is not in the document]
Comment (pietro909) [reply ts/screens/messages/paginated/MigratingMessage.tsx:43]: Fixed [here](https://github.com/pagopa/io-app/pull/3853/commits/bb38a9404402fced607de1b5193023146163cb24#diff-ced3d0535c496b8124b980c773e751099334a086e942481b881083794bdd34f4L44)

File: locales/it/index.yml [hunk 3]
```diff
@@ -1700,6 +1700,11 @@ messages:
     barcodes: Codici a barre
     nationalService: Servizio Sanitario Nazionale
     note: "Rilasciato ai sensi dell'art.11, comma 16 del DL 31 mag 2010, n.78 e dell’art.1, comma 4 del DM 2 nov 2011"
+  pagination:
+    migration:
+      started: "Attendi per favore, stiamo aggiornando la pagina dei messaggi"
+      succeeded: "Abbiamo aggiornato la pagina dei messaggi!"
+      failed: "Non siamo riusciti ad aggiornare la pagina dei messaggi a causa di un errore. Premi il bottone per riprovare"
```
[the remark this answers is not in the document]
Comment (pietro909) [reply locales/it/index.yml:1707]: @thisisjp is working on this.

**Review by emiliopavia — APPROVED** [review]
In my tests it seems to work as described.
